src/input/gamepad/gamepadCalibration.js

```diff
@@ -0,0 +1,500 @@
+// @flow
+/*eslint indent:0*/
+
+import {Vec2D} from "../../main/util/Vec2D";
+import {deepCopyObject} from "../../main/util/deepCopyObject";
+import {setCustomGamepadInfo} from "./gamepads/custom";
+import {setUsingCustomControls, currentPlayers} from "../../main/main";
+import {updateControllerMenu} from "../../menus/controllermenu.js";
+import {nullGamepadInfo} from "./gamepadInfo";
+import {getGamepad} from "./gamepad";
+
+// eslint-disable-next-line no-duplicate-imports
+import type {ButtonInfo, StickInfo, StickCardinals, TriggerInfo, DPadInfo, GamepadInfo} from "./gamepadInfo";
+// eslint-disable-next-line no-duplicate-imports
+import type {Gamepad, Button} from "./gamepad";
+
+
+const calibrationInProgress = [false, false, false, false];
+function setCalibrationInProgress (i : number, bool : bool) : void {
+  calibrationInProgress[i] = bool;
+}
+
+function deepCopyButtons( gpd : Gamepad ) : Array<Button> {
+  const lgB = gpd.buttons.length;
```

Comment:
you can probably get away with using deepCopyObject(false.......) since its just an alias for Object.assign()